nupic/algorithms/CLAClassifier.py

```diff
@@ -213,6 +213,31 @@ def __setstate__(self, state):
     self._version = BitHistory.__VERSION__
 
 
+  def write(self, proto):
+    proto.id = self._id
+
+    statsProto = proto.init("stats", len(self._stats))
+    for (bucketIdx, dutyCycle) in enumerate(self._stats):
+      statsProto[bucketIdx].index = bucketIdx
+      statsProto[bucketIdx].dutyCycle = dutyCycle
+
+    proto.lastTotalUpdate = self._lastTotalUpdate
+    proto.learnIteration = self._learnIteration
+
+
+  def read(self, proto):
```

Comment:
The original suggestion to use `__new__` as opposed to `__init__` was deliberate.  There is overlap in functionality here between `CLAClassifier.__init__()` and `CLAClassifier.read()`.  In Python, `__init__()` is not a traditional _constructor_, it _initializes_ an already existing _instance_ (created with `object.__new__(cls)`).

Since we're working with serialization, the object was already technically initialized before it was serialized, and then by calling `read()`, you're re-initing the object and then overwriting the values from the serialized object.